internal/service/lambda/provisioned_concurrency_config_test.go

```diff
@@ -159,9 +163,40 @@ func TestAccLambdaProvisionedConcurrencyConfig_Qualifier_aliasName(t *testing.T)
 				),
 			},
 			{
-				ResourceName:      resourceName,
-				ImportState:       true,
-				ImportStateVerify: true,
+				ResourceName:            resourceName,
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"skip_destroy"},
+			},
+		},
+	})
+}
+
+func TestAccLambdaProvisionedConcurrencyConfig_skipDestroy(t *testing.T) {
+	ctx := acctest.Context(t)
+	if os.Getenv("SKIP_DESTROY_ENABLED") == "" {
+		t.Skip("Environment variable SKIP_DESTROY_ENABLED is not set")
+	}
+
+	rName := sdkacctest.RandomWithPrefix(acctest.ResourcePrefix)
+	lambdaFunctionResourceName := "aws_lambda_function.test"
+	resourceName := "aws_lambda_provisioned_concurrency_config.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:                 func() { acctest.PreCheck(ctx, t) },
+		ErrorCheck:               acctest.ErrorCheck(t, names.LambdaEndpointID),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories,
+		CheckDestroy:             testAccCheckProvisionedConcurrencyConfigNoDestroy(ctx),
+		Steps: []resource.TestStep{
```

Comment:
We could test this without the environment variable by using the following test steps, which would also test the originally reported problem

1. Create
2. Publish a new version of the function, but validate that the old version of the function still has concurrency set
3. Delete if from the new version, and validate that it's still set on the new version

Reply:
Good suggestion - I've updated to this pattern and we shouldn't have any dangling resources leftover now.